tests/fixtures/padDifference.ts

```diff
@@ -0,0 +1,264 @@
+import { footprintSizes } from "src/helpers/passive-fn"
+import { transformPcbElements } from "@tscircuit/circuit-json-util"
+import { translate } from "transformation-matrix"
+import { fp } from "src/footprinter"
+import type { PcbSilkscreenText } from "circuit-json"
+
+type PcbSmtPad = {
+  type: "pcb_smtpad"
+  x: number
+  y: number
+  width: number
+  height: number
+  port_hints: string[]
+}
+
+type PcbComponent = {
+  type: "pcb_component"
+  width: number
+  height: number
+  center: { x: number; y: number }
+}
+
+const safe = (v: number | undefined) => (typeof v === "number" ? v : 0)
+
+function relativeDiffToRef(kicadVal: number, refVal: number) {
+  if (refVal === 0) return 0
+  return Math.abs(kicadVal - refVal) / refVal
+}
+
+export async function padDifference(
+  imperialOrMetric: string,
+  footprintName: string,
+): Promise<{
+  avgRelDiff: number
+  combinedFootprintElements: any[]
+}> {
+  const normalized = imperialOrMetric.replace(/_(imp|metric)$/, "")
```

Comment:
None of this logic should be here, you should just render the footprint to circuit json, then compute what area is covered by smtpads

```suggestion
```